Feature/6 account creation nextjs

ifirmawan opened the pull request:
## TODO/DONE

- [x] Use image placeholder for now
- [x]  try to reproduce the layout only and use the correct colours
- [x] Use ISO 3166 countries. 
- [x] Setup frontend internationalisation for french and english including translation files. 
- [x] Use email validation
- [x] Use password rules
- [x] Pop-up PAT Terms

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1208203113907817

ifirmawan committed:
[#6] Create i18n dir to store translations & countries

ifirmawan committed:
[#6] Attach i18n to backend & frontend

ifirmawan committed:
[#6] Install antd, antd/nextjs-register, classNames & next-intl

ifirmawan committed:
[#6] Setup frontend i18n for french and english

ifirmawan committed:
[#6] Move homepage including tests to locale route dir

ifirmawan committed:
[#6] Handle 404 page

ifirmawan committed:
[#6] Update jest config to handle next-intl

ifirmawan committed:
[#6] Customize initial fonts, color & antd form

ifirmawan committed:
[#6] Create static config files

ifirmawan committed:
[#6] Create new components: Icons

ifirmawan committed:
[#6] Create RegisterForm component

ifirmawan committed:
[#6] Create auth template page

ifirmawan committed:
[#6] Create a new page: register

ifirmawan committed:
[#6] Create initial page: Login for redirection purpose

ifirmawan committed:
[#6] Update password strength modal title

ifirmawan committed:
[#6] Move password strength & add rules: gender & country

dedenbangkit commented on frontend/src/components/Forms/RegisterForm.js:
```diff
@@ -0,0 +1,257 @@
+          value={checkedList}
+        />
+      </Modal>
+      <Form.Item
```
This should be disabled if password criteria didn't pass

dedenbangkit commented on frontend/src/components/Forms/RegisterForm.js:
```diff
@@ -0,0 +1,257 @@
+              if (checkedList.length === 6) {
+                return Promise.resolve();
+              }
+              return Promise.reject(new Error(tc("passwordCriteriaError")));
```
Should inform the error in detail (add info icon) after "passwordCriteriaError" with tooltip: list of criteria.

dedenbangkit commented on frontend/src/components/Forms/RegisterForm.js:
```diff
@@ -0,0 +1,257 @@
+        />
+      </div>
+
+      <Button type="primary" htmlType="submit" disabled={!isChecked} block>
```
disabled is not only isChecked, but `&&` everything is complete

dedenbangkit commented on frontend/src/components/Forms/RegisterForm.js:
```diff
@@ -0,0 +1,257 @@
+  const checkBoxOptions = [
+    { name: tc("passwordRule1"), re: /[a-z]/ },
+    { name: tc("passwordRule2"), re: /\d/ },
+    {
```
Skip symbols from the password rules

ifirmawan committed:
[#6] Add new translations: checkAgreemntRequired

ifirmawan committed:
[#6] Handle input disabled styles

ifirmawan committed:
[#6] Create SubmitButton component that watch all input rules

ifirmawan committed:
[#6] Fix register form fields check

- Disabled confirm password when the password criteria not match
- Implements Submit button to fix feedback "disabled is not only isChecked, but && everything is complete"

ifirmawan committed:
[#6] Skip symbols from the password rules

ifirmawan committed:
[#6] Handle non feedback input group styles

ifirmawan committed:
[#6] Add new icon: WarningCircle

ifirmawan committed:
[#6] Inform the error in detail (add info icon)

Add info icon after "passwordCriteriaError" with tooltip: list of criteria

ifirmawan committed:
[#6] Fix responsive issue template in tab & mobile view
